app/api/migrations/migrations/36-populate-mimetype-on-attachments/index.js

```diff
@@ -0,0 +1,34 @@
+import request from 'shared/JSONRequest';
+import { attachmentsPath } from 'api/files/filesystem';
+import { execSync } from 'child_process';
+
+export default {
+  delta: 36,
+
+  name: 'populate-mimetype-to-attachment',
+
+  description: 'Populates mimetype of an attachment from a url ',
+
+  async up(db) {
+    process.stdout.write(`${this.name}...\r\n`);
+    const cursor = await db.collection('files').find({});
+
+    // eslint-disable-next-line no-await-in-loop
+    while (await cursor.hasNext()) {
+      // eslint-disable-next-line no-await-in-loop
+      const file = await cursor.next();
+      if (file.url && !file.mimetype) {
+        // eslint-disable-next-line no-await-in-loop
+        const response = await request.head(file.url);
+        const mimetype = response.headers.get('content-type') || undefined;
+        db.collection('files').updateOne({ _id: file._id }, { $set: { mimetype } });
+      }
+      if (file.filename && file.type === 'attachment' && !file.mimetype) {
```

Comment:
Does it make sense that this `if` is in `else` with the previous?

Reply:
Kindly explain more @fnocetti

Reply:
Added